utils/update_build_version.py

```diff
@@ -63,85 +71,111 @@ def command_output(cmd, directory):
+    if not success:
+      return False, None
+
+    latest_version_tag = None
+    for tag in tag_list.decode().splitlines():
+      if VERSION_REGEX.match(tag):
+        latest_version_tag = tag
+        break
+
+    if latest_version_tag is None:
+      logging.error("No tag matching version regex matching.")
+      return False, None
+    return True, latest_version_tag
+
+def get_description_for_head(repo_path):
     """Returns a string describing the current Git HEAD version as descriptively
     as possible.
 
     Runs 'git describe', or alternately 'git rev-parse HEAD', in directory.  If
     successful, returns the output; otherwise returns 'unknown hash, <date>'."""
-    try:
-        # decode() is needed here for Python3 compatibility. In Python2,
-        # str and bytes are the same type, but not in Python3.
-        # Popen.communicate() returns a bytes instance, which needs to be
-        # decoded into text data first in Python3. And this decode() won't
-        # hurt Python2.
-        return command_output(['git', 'describe'], directory).rstrip().decode()
-    except:
-        try:
-            return command_output(
-                ['git', 'rev-parse', 'HEAD'], directory).rstrip().decode()
-        except:
-            # This is the fallback case where git gives us no information,
-            # e.g. because the source tree might not be in a git tree.
-            # In this case, usually use a timestamp.  However, to ensure
-            # reproducible builds, allow the builder to override the wall
-            # clock time with environment variable SOURCE_DATE_EPOCH
-            # containing a (presumably) fixed timestamp.
-            timestamp = int(os.environ.get('SOURCE_DATE_EPOCH', time.time()))
-            formatted = datetime.datetime.utcfromtimestamp(timestamp).isoformat()
-            return 'unknown hash, {}'.format(formatted)
 
+    success, output = command_output(['git', 'describe'], repo_path)
+    if not success:
+      output = command_output(['git', 'rev-parse', 'HEAD'], repo_path)
+
+    if success:
+      # decode() is needed here for Python3 compatibility. In Python2,
+      # str and bytes are the same type, but not in Python3.
+      # Popen.communicate() returns a bytes instance, which needs to be
+      # decoded into text data first in Python3. And this decode() won't
+      # hurt Python2.
+      return output.rstrip().decode()
+
+    # This is the fallback case where git gives us no information,
+    # e.g. because the source tree might not be in a git tree.
+    # In this case, usually use a timestamp.  However, to ensure
+    # reproducible builds, allow the builder to override the wall
+    # clock time with environment variable SOURCE_DATE_EPOCH
+    # containing a (presumably) fixed timestamp.
+    timestamp = int(os.environ.get('SOURCE_DATE_EPOCH', time.time()))
+    iso_date = datetime.datetime.utcfromtimestamp(timestamp).isoformat()
+    return "unknown hash, {}".format(iso_date)
```

Comment:
This seems to be a refactoring. This might be better in a separate PR from the actual change of behaviour. Smaller PRs make it easier to review and debug.